app/controllers/journeys/expenses/tailoring/ProfessionalServiceExpensesController.scala

```diff
@@ -17,53 +17,76 @@
+    override val messagesApi: MessagesApi,
+    selfEmploymentService: SelfEmploymentService,
+    sessionRepository: SessionRepository,
+    navigator: ExpensesTailoringNavigator,
+    identify: IdentifierAction,
+    getData: DataRetrievalAction,
+    formProvider: ProfessionalServiceExpensesFormProvider,
+    val controllerComponents: MessagesControllerComponents,
+    view: ProfessionalServiceExpensesView
+)(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport {
 
-  val form = formProvider()
+  val businessId = "SJPR05893938418"
 
-  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData) { implicit request =>
-    val preparedForm = request.userAnswers.getOrElse(UserAnswers(request.userId)).get(ProfessionalServiceExpensesPage) match {
-      case None        => form
-      case Some(value) => form.fill(value)
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData) async { implicit request =>
+    selfEmploymentService.getAccountingType(request.user.nino, businessId, request.user.mtditid) map {
+      case Left(_) => Redirect(JourneyRecoveryController.onPageLoad())
+      case Right(accountingType) =>
+    val preparedForm = request.userAnswers.getOrElse(UserAnswers(request.userId)).get(ProfessionalServiceExpensesPage, Some(businessId)) match {
+      case None => formProvider(userType(request.user.isAgent))
+      case Some(value) => formProvider(userType(request.user.isAgent)
+      ).fill(value)
     }
 
-    Ok(view(preparedForm, mode))
+    Ok(view(preparedForm, mode, userType(request.user.isAgent), accountingType, legendContent(userType(request.user.isAgent))))
   }
+}
 
   def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData) async { implicit request =>
-    form
+    selfEmploymentService.getAccountingType(request.user.nino, businessId, request.user.mtditid) flatMap {
+      case Left(_) => Future.successful(Redirect(JourneyRecoveryController.onPageLoad()))
+      case Right(accountingType) =>
+    formProvider(userType(request.user.isAgent))
       .bindFromRequest()
       .fold(
-        formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode, authUserType = userType(request.user.isAgent),  accountingType,
+          legendContent(userType(request.user.isAgent))))),
         value =>
           for {
             updatedAnswers <- Future.fromTry(request.userAnswers.getOrElse(UserAnswers(request.userId)).set(ProfessionalServiceExpensesPage, value))
-            _              <- sessionRepository.set(updatedAnswers)
+            _ <- sessionRepository.set(updatedAnswers)
           } yield Redirect(navigator.nextPage(ProfessionalServiceExpensesPage, mode, updatedAnswers))
       )
   }
+  }
+
+  private def legendContent(userType: String)(implicit messages: Messages) = buildLegendHeadingWithHintString(
```

Comment:
We deleted this method in https://github.com/hmrc/income-tax-self-employment-frontend/pull/74 because the logic should be in the view not the controller. You can copy the format in the PR and then delete the ContentStringViewModel class so that merging your PR doesn't remake it